db/repos/eventsRepository.js

```diff
@@ -0,0 +1,120 @@
+const { NotFound } = require('../../services/errors');
+
+class EventsRepository {
+  constructor(db, userRepository) {
+    this.db = db;
+    this.userRepository = userRepository;
+  }
+
+  async findAllEvents(params) {
+    const { Event, User } = this.db;
+
+    const query = {
+      attributes: this.getOwnAttributes(),
+      include: [
+        {
+          model: User,
+          as: 'participants',
+          attributes: this.userRepository.getOwnAttributes(),
+        },
+      ],
+    };
+
+    if (params.location) {
+      query.where = {
+        location: query.location,
+      };
+    }
+
+    return await Event.findAll(query);
+  }
+
+  async getEvent(id) {
+    const { Event, User } = this.db;
+    const query = {
+      attributes: this.getOwnAttributes(),
+      include: [
+        {
+          model: User,
+          as: 'participants',
+          attributes: this.userRepository.getOwnAttributes(),
+        },
+      ],
+      where: {
+        id,
+      },
+    };
+
+    return await Event.findOne(query);
+  }
+
+  async delete(id) {
+    const { Event } = this.db;
+
+    return await Event.destroy({ where: { id } });
+  }
+
+  async create(data) {
+    const { Event, User, UserEvent, sequelize } = this.db;
+    const { title, location, date, creatorId, participants } = data;
+
+    let transaction;
```

Comment:
you could write `const  transaction = await sequelize.transaction();` here and it'd work fine. Try to avoid usage of `let`. This always leads to unexpected errors.